Sources/TuistGenerator/Dot/DotGraphGenerator.swift

```diff
@@ -0,0 +1,72 @@
+import Basic
+import Foundation
+import TuistCore
+
+public protocol DotGraphGenerating {
+    /// Generates the dot graph from the project in the current directory and returns it.
+    ///
+    /// - Parameter path: Path to the folder that contains the project.
+    /// - Returns: Dot graph representation.
+    /// - Throws: An error if the project can't be loaded.
+    func generateProject(at path: AbsolutePath) throws -> String
+
+    /// Generates the dot graph from the workspace in the current directory and returns it.
+    ///
+    /// - Parameter path: Path to the folder that contains the workspace.
+    /// - Returns: Dot graph representation.
+    /// - Throws: An error if the workspace can't be loaded.
+    func generateWorkspace(at path: AbsolutePath) throws -> String
+}
+
+public final class DotGraphGenerator: DotGraphGenerating {
+    /// Graph loader.
+    private let graphLoader: GraphLoading
+
+    /// Mapper to map graphs into a dot graphs.
+    private let graphToDotGraphMapper: GraphToDotGraphMapping
+
+    /// Initializes the dot graph generator by taking its dependencies.
+    ///
+    /// - Parameters:
+    ///   - modelLoader: Instance to load the models.
+    ///   - printer: Instance to print outputs to the user.
+    ///   - fileHandler: Instance to handle files.
+    public convenience init(modelLoader: GeneratorModelLoading,
+                            printer: Printing = Printer(),
+                            fileHandler: FileHandling = FileHandler()) {
+        let graphLinter = GraphLinter(fileHandler: fileHandler)
+        let graphLoader = GraphLoader(linter: graphLinter, printer: printer, fileHandler: fileHandler, modelLoader: modelLoader)
```

Comment:
I haven't thought about it. I agree with not linting here. If a dependency is wrong, the graph loading will fail and developers will get an error in the terminal. I think linting makes more sense when generating the project because in that case, we want to make sure that developers don't get a project that might not compile.